Sources/Controllers/2.0/OACustomPOIViewController.mm

```diff
@@ -68,39 +99,77 @@ - (void)viewDidLoad
 
     self.tableView.delegate = self;
     self.tableView.dataSource = self;
-    self.tableView.tableHeaderView = [OAUtilities setupTableHeaderViewWithText:OALocalizedString(@"search_poi_types_descr") font:kHeaderViewFont textColor:UIColorFromRGB(color_text_footer) lineSpacing:6.0 isTitle:NO];
+    [self.tableView registerClass:OATableViewCustomHeaderView.class forHeaderFooterViewReuseIdentifier:kHeaderId];
+
+    _searchMode = NO;
+    self.searchBar.delegate = self;
+    UITextField *textField = [self.searchBar valueForKey:@"searchField"];
```

Comment:
Make sure this is not prohibited by Apple and will not get us banned from the appstore